timemachine/md/moves.py

```diff
@@ -16,16 +16,18 @@
 from timemachine.md.states import CoordsVelBox
 from timemachine.potentials import BoundPotential, HarmonicBond
 
+_State = TypeVar("_State")
 
-class MonteCarloMove:
+
+class MonteCarloMove(Generic[_State]):
     n_proposed: int = 0
     n_accepted: int = 0
 
-    def propose(self, x: CoordsVelBox) -> Tuple[CoordsVelBox, float]:
+    def propose(self, x: _State) -> Tuple[_State, float]:
         """return proposed state and log acceptance probability"""
         raise NotImplementedError
 
-    def move(self, x: CoordsVelBox) -> CoordsVelBox:
+    def move(self, x: _State) -> _State:
```

Comment:
Nice -- this is useful.

In addition to the motivating example of HREX, other cases that indicate "state = (x,v,b)" was prematurely specialized include:
* cases where the state omits velocities (instantaneous displacement moves)
* cases where the move cannot update box (NVT)
* cases where the move is allowed to update other auxiliary variables (`(x, box, selection_mask)`, `(x, v, box, selection_mask)` , etc. in localized moves)
* cases where the move only updates a subset of coordinates